docs/rfcs/rfc-010-top-level-interfaces.md

```diff
@@ -0,0 +1,138 @@
+# Top-level Interfaces
+
+## Problem
+
+Interface types are currently only available to Query as part of a relationship field. 
+Our users would like to be able to query for Interface types directly from the root Query type.
+
+## Proposed Solution
+
+Given the following type definitions:
+
+```gql
+interface Production {
+  title: String!
+}
+
+type Movie implements Production {
+  title: String!
+}
+
+type Series implements Production {
+  title: String!
+}
+```
+
+We would expect to see the following new field in the Query root type:
+
+```gql
+type Query {
+  productions(options: ProductionOptions, where: ProductionWhere): [Production!]!
+}
+```
+
+### Usage Examples
+
+Using the type definitions above, let's look at some examples.
+
+```gql
+query {
+  productions {
+    title
+  }
+}
+```
+
+This would translate into:
+
+```
+CALL {
+  MATCH (this:Movie)
+  RETURN this { __resolveType: "Movie", .title }
+UNION
+  MATCH (this:Series)
+  RETURN this { __resolveType: "Series", .title }
+}
+RETURN this
+```
+
+We can do some filtering for both types:
+
+```gql
+query {
+  productions(where: { _on: { Movie: { title: "Forrest Gump" }, Series: { title: "Star Wars: The Clone Wars" } } }) {
```

Comment:
Would it be possible to filter directly on `Production`? , if so, an example would be useful

something a bit like:

```graphql
production(where: {title: "Start Wars" })
```